src/winrtble/manager.rs

```diff
@@ -31,15 +31,12 @@ impl api::Manager for Manager {
     type Adapter = Adapter;
 
     async fn adapters(&self) -> Result<Vec<Adapter>> {
-        let mut result: Vec<Adapter> = vec![];
-        let radios = Radio::GetRadiosAsync().unwrap().await.unwrap();
-
-        for radio in &radios {
-            let kind = radio.Kind().unwrap();
-            if kind == RadioKind::Bluetooth {
-                result.push(Adapter::new());
-            }
-        }
-        return Ok(result);
+        let radios = Radio::GetRadiosAsync()?.await?;
+        Ok(radios
+            .into_iter()
+            .filter_map(|radio| radio.Kind().ok())
+            .filter(|&kind| kind == RadioKind::Bluetooth)
```

Comment:
Nit: could you do this in one filter, like `filter(|radio|radio.Kind() == Ok(RadioKind::Bluetooth))`?

Reply:
Updated. Thanks for the hint.